samples/websocket_connect_custom_auth.md

```diff
@@ -0,0 +1,107 @@
+On startup, the device connects to the server and then disconnects.
+This sample is for reference on connecting using a Custom Authorizer.
+Using a Custom Authorizer allows you to perform your own authorization using an AWS Lambda function.
+See [Custom Authorizer](https://docs.aws.amazon.com/iot/latest/developerguide/custom-authentication.html) for more information.
+You will need to setup your Custom Authorizer so that the lambda function returns a policy document.
+See [this page on the documentation](https://docs.aws.amazon.com/iot/latest/developerguide/config-custom-auth.html) for more details and example return result.
+You can customize this lambda function as needed for your application to provide your own security measures based on the needs of your application.
+Your IoT Core Thing's [Policy](https://docs.aws.amazon.com/iot/latest/developerguide/iot-policies.html) must provide privileges for this sample to connect.
+Below is a sample policy that can be used on your IoT Core Thing that will allow this sample to run as intended.
+
+If you want to use simple or custom auth (or static creds, or basic auth, etc) instead,
+then you will need to replace part of the sample (connection\_setup function) with a code snippet we provided in its corresponding readme.
+
+Your IoT Core Thing's [Policy](https://docs.aws.amazon.com/iot/latest/developerguide/iot-policies.html) must provide privileges for this sample to connect. Below is a sample policy that can be used on your IoT Core Thing that will allow this sample to run as intended.
+
+<details>
+<summary>(see sample policy)</summary>
+<pre>
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "iot:Connect"
+      ],
+      "Resource": [
+        "arn:aws:iot:<b>region</b>:<b>account</b>:client/test-*"
+      ]
+    }
+  ]
+}
+</pre>
+
+
+Replace with the following with the data from your AWS account:
+* `<region>`: The AWS IoT Core region where you created your AWS IoT Core thing you wish to use with this sample. For example `us-east-1`.
+* `<account>`: Your AWS IoT Core account ID. This is the set of numbers in the top right next to your AWS account name when using the AWS IoT Core website.
+
+Note that in a real application, you may want to avoid the use of wildcards in your ClientID or use them selectively. Please follow best practices when working with AWS on production applications using the SDK. Also, for the purposes of this sample, please make sure your policy allows a client ID of `test-*` to connect or use `--client_id <client ID here>` to send the client ID your policy supports.
+
+For this sample, using Websockets will attempt to connect using custom auth.
+
+</details>
+
+<details>
+<summary> (code snipet to replace similar section)</summary>
+<pre language="python">
+<code>
+def connection_setup():
```

Comment:
there is no `def connection_setup()`  in the websocket_connect.py. `connection_setup()` is also not called in the main function of websocket_connect.py. Let's change both these snippets to be more descriptive of what is being changed and what it's doing.

e.g. first code snippet, show what is being changed and why
This will parse the command line argument to run the sample for the arguments for connecting to the websocket using a custom authorizer.
 code block
# cmdData = CommandLineUtils.parse_sample_input_websocket_connect()
cmdData = CommandLineUtils.parse_sample_input_custom_authorizer_connect()
 /code block

Secondary code snippet that describes and shows changes in main

Reply:
Could we also add remarkable comments to highlight where/what to change in `websocket_connect.py` .
For example: 

```python
################ BEGIN SECTION ###############
## Replace this section to parse different arguments #### 

cmdData = CommandLineUtils.parse_sample_input_custom_authorizer_connect()

################ BEGIN SECTION ###############

```

Reply:
no need to, as the user has to just replace the function connection_setup and nothing else

Reply:
I didnt find `connection_setup()` in `samples/websocket_connect.py`, while I did see it in our cpp samples. Are we missing the update for Python?

Reply:
added